src/test/java/org/wildfly/security/sasl/oauth2/OAuth2SaslClientTest.java

```diff
@@ -158,8 +157,8 @@ public void failedWithBearerTokenFromConfiguration() throws Exception {
         try {
             message = saslServer.evaluateResponse(message);
         } catch (SaslException e) {
-            assertTrue(e.getCause() instanceof RealmUnavailableException);
-            RealmUnavailableException cause = (RealmUnavailableException) e.getCause();
+            assertTrue(e.getCause() instanceof IllegalArgumentException);
```

Comment:
We are already logging messages when specific events occur, specially those events related with validations that end up in an exception being thrown. For instance, signature checks, issuer checks, bad communication channel with an OAuth2 introspection endpoint, etc.

Is this OK ?